lib/pure/osproc.nim

```diff
@@ -1387,22 +1388,26 @@ elif not defined(useNimRtl):
             raiseOsError(osLastError())
           else:
             # Continue waiting if needed
-            if getTime() >= deadline:
+            if getMonoTime() >= deadline:
               # Previous version of `waitForExit`
               # foricibly killed the process.
               # We keep this so we don't break programs
               # that depend on this behavior
               if posix.kill(p.id, SIGKILL) < 0:
                 raiseOSError(osLastError())
             else:
-              let newWait = getTime() + delay
+              const max = 999_999_999
+              let 
+                newWait = getMonoTime() + delay
+                ns = newWait.ticks() mod max
+                secs = (newWait.ticks() - ns) div 1_000_000_000
```

Comment:
```nim
                ns = newWait.ticks() mod 1_000_000_000
                secs = newWait.ticks() div 1_000_000_000
```

Reply:
Put `newWait.tick()` in a local variable and reuse it.